### ISSUE TITLE
[DatePicker] Days of week are marked aria-hidden making them inaccessible to screen readers

### ISSUE THREAD

**Issue by slayybelle** [mui/mui-x#4470]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Location: Second row of Date Picker where days of the week are listed "S M T W T F S" are all marked aria-hidden:

<span class="MuiTypography-root MuiTypography-caption css-10qkmc3" aria-hidden="true">S</span>

### Expected behavior 🤔

Columns for day of week are relevant information so should not be marked aria-hidden.  This was flagged as a "critical" issue (one below "Blocker") by our a11y auditors.

Could you pls remove the aria-hidden attribute?

Here's where it is in the code:
<img width="799" alt="Screen Shot 2022-02-15 at 8 54 03 AM" src="https://user-images.githubusercontent.com/32496958/154110021-81786bbf-0209-4c0c-aef7-44c39877ff6e.png">
 

### Steps to reproduce 🕹

Steps:

1. Go to Date picker demo page "Basic Usage":  https://mui.com/components/date-picker/#basic-usage
2. Open up devtools and select any one of "S M T W T F S" 
3.  You'll see the aria-hidden attribute
<img width="1388" alt="Screen Shot 2022-02-15 at 8 57 30 AM" src="https://user-images.githubusercontent.com/32496958/154110590-35f8a0aa-6c04-4dfa-83fc-2188088246b5.png">

### Context 🔦

In middle of a11y audit by 3rd party reputable a11y firm. 

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by LukasTy** [comment]
Hello @slayybelle 
Could you by any chance test the [following changes](https://deploy-preview-5809--material-ui-x.netlify.app/x/react-date-pickers/date-picker/) and see if a11y support in date pickers is improved?
You should be able to test the components locally by using the [pre-release package](https://pkg.csb.dev/mui/mui-x/commit/d7788b74/@mui/x-date-pickers)

**Comment by slayybelle** [comment]
VoiceOver announces the days of the week ✅ 

Some other observations:
The aria-label of date cells in current released picker includes the entire date, ex "August 17, 2022" whereas this new one only has a number "17". it's better to include the entire date.  It can be confusing if you up arrow into another month but at least in the current released version you will know right away because the label includes "month day, year"

I manually added "aria-current" to today's date (the day cell with the circle around the date); VoiceOver announced this.  You may consider adding aria-current for cell denoting today's date.

I noticed aria-selected has been added. ✅  even with aria-selected="true" on the selected date, VO did not announce this state, but I'm thinking this is a VO bug and it's good that you've added this. I just fired up VMware windows with NVDA.  The aria-selected="false" cells were announced as "not selected" but she never said "selected" for the one that was selected. Weird but better than VO

Sorry, not sure what else you added with this PR.  I know there are still issues with keyboard focus going back to the date cells if the selected cell is not present in the current view; it feels as if the code governing keyboard focus needs the selected cell to focus and the absence of this cell prevents focus from entering the date cell area.  For example, if my currently selected date is Sept 20 and I am viewing August, there is no way to enter the date area but when I switch to September, now my focus is redirected to Sept 20 cell and I can move around the date cells.  I ramble... 

Thx for adding more a11y to this module!

**Comment by LukasTy** [comment]
Thanking for having a look at this. 🙏 

I'll preface it by saying that I basically ignored how VO works as it probably does not support half of the a11y features. As far as I know, industry standard for visually impaired users is JAWS or NVDA on windows, both having a pretty good support.
The issue is that JAWS is a paid tool and doesn't work on ARM (Apple Silicon).

> The aria-label of date cells in current released picker includes the entire date, ex "August 17, 2022" whereas this new one only has a number "17". it's better to include the entire date. It can be confusing if you up arrow into another month but at least in the current released version you will know right away because the label includes "month day, year"

Including the whole date in the label is sort of a hacky approach to achieve screen reader a11y.
If we were to add both the `aria-label` on date and correct `aria-live` updating on `Moth Year` header section, we would get the double announcement of same data. Also, I could say, that I found it counter intuitive hearing the same year and month on every day navigation, whereas in the new version (and W3-ARIA) examples heading with `aria-live` is used to announce the change of month or year to the user once.

> I manually added "aria-current" to today's date (the day cell with the circle around the date); VoiceOver announced this. You may consider adding aria-current for cell denoting today's date.

Thanks for this note, I'll look into it.

> I noticed aria-selected has been added...

Yes, this implementation is buggy in NVDA as far as I can say, because adding `aria-selected` as per W3-ARIA suggestion (`aria-selected="true"` only on the selected cell) causes weird behaviour, where `selected` is announced when selecting any **not** selected date, that why I opted for having `aria-selected` on all cells, which hopefully works as intended on JAWS. 🤞 

> Sorry, not sure what else you added with this PR. I know there are still issues with keyboard focus going back to the date cells if the selected cell is not present in the current view

The keyboard focus issues are [being worked on another branch](https://github.com/mui/mui-x/pull/5820) by a colleague. 😉
